### ISSUE TITLE
Legion bug

### ISSUE THREAD

**Issue by ceylina** [jrob8577/Auctionator#68]
As I left the message on slack, here is the error I get trying to put something on the sell tab. Looks like a few more tweaks are needed to item classes.

Buy seems to work great and no errors on launch or reload so super impressed you patched something together yesterday.

Auctionator\AuctionatorLocalize.lua:354: attempt to get length of local 'itemClasses' (a nil value)
Auctionator\AuctionatorLocalize.lua:354: in function `Atr_ItemType2AuctionClass'
Auctionator\AuctionatorLocalize.lua:295: in function `Atr_IsClass'
Auctionator\AuctionatorLocalize.lua:270: in function `Atr_IsGlyph'
Auctionator\Auctionator-4.0.0.lua:4420: in function `Atr_Special_SP_ByItem'
Auctionator\Auctionator-4.0.0.lua:4374: in function `Atr_GetStackingPrefs_ByItem'
Auctionator\Auctionator-4.0.0.lua:4453: in function `Atr_GetSellStacking'
Auctionator\Auctionator-4.0.0.lua:3000: in function `Atr_OnNewAuctionUpdate'
Auctionator\Auctionator-4.0.0.lua:194: in function `Atr_EventHandler'
[string "*:OnEvent"]:1: in function <[string "*:OnEvent"]:1>
[C]: in function `ClickAuctionSellItemButton'
Auctionator\Auctionator-4.0.0.lua:1548: in function `Atr_ClickAuctionSellItemButton'
Auctionator\Auctionator-4.0.0.lua:1565: in function `Atr_OnDropItem'
[string "*:OnReceiveDrag"]:1: in function <[string "*:OnReceiveDrag"]:1>

Locals:
itemType = "Tradeskill"
itemClasses = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = "Atr_ItemType2AuctionClass"
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = "attempt to get length of local 'itemClasses' (a nil value)"
zc = <table> {
 msg_anm = <function> defined @Auctionator\zcUtils.lua:624
 msg = <function> defined @Auctionator\zcUtils.lua:724
 GetArrayElemOrFirst = <function> defined @Auctionator\zcUtils.lua:161
 NumToBool = <function> defined @Auctionator\zcUtils.lua:549
 StringStartsWith = <function> defined @Auctionator\zcUtils.lua:917
 StringSame = <function> defined @Auctionator\zcUtils.lua:861
 padstring = <function> defined @Auctionator\zcUtils.lua:191
 GetArrayElemOrNil = <function> defined @Auctionator\zcUtils.lua:176
 BoolToNum = <function> defined @Auctionator\zcUtils.lua:539
 periodic = <function> defined @Auctionator\zcUtils.lua:368
 pluralizeIf = <function> defined @Auctionator\zcUtils.lua:570
 msg_pink = <function> defined @Auctionator\zcUtils.lua:596
 SetTextIf = <function> defined @Auctionator\zcUtils.lua:47
 HSV2RGB = <function> defined @Auctionator\zcUtils.lua:639
 LinkFromItemID = <function> defined @Auctionator\zcUtils.lua:492
 IsBattlePetLink = <function> defined @Auctionator\zcUtils.lua:434
 printableLink = <function> defined @Auctionator\zcUtils.lua:1030
 PrintTable = <function> defined @Auctionator\zcUtils.lua:398
 PrintKeysSorted = <function> defined @Auctionator\zcUtils.lua:121
 Negate = <function> defined @Auctionator\zcUtils.lua:559
 IsTextQuoted = <function> defined @Auctionator\zcUtils.lua:941
 CopyDeep = <function> defined @Auctionator\zcUtils.lua:1009
 UTF8_Truncate = <function> defined @Auctionator\zcUtils.lua:139
 val2gsc = <function> defined @Auctionator\zcUtils.lua:774
 words = <function> defined @Auctionator\zcUtils.lua:297
 tableIsEmpty = <function> defined @Auctionator\zcUtils.lua:386
 Max = <function> defined @Auctionator\zcUtils.lua:95
 printstack = <function> defined @Auctionator\zcUtils.lua:1054
 pluralize = <function> defined @Auctionator\zcUtils.lua:581
 RawItemIDfromLink = <function> defined @Auctionator\zcUtils.lua:462
 EnableDisable = <function> defined @Auctionator\zcUtils.lua:25
 TrimBrackets = <function> defined @Auctionator\zcUtils.lua:980
 ParseBattlePetLink = <function> defined @Auctionator\zcUtils.lua:442
 ShowHide = <function> defined @Auctionator\zcUtils.lua:36
 QuoteString = <function> defined @Auctionator\zcUtils.lua:948
 round = <function> defined @Auctionator\zcUtils.lua:589
 StringEndsWith = <function> defined @Auctionator\zcUtils.lua:897
 priceToMoneyString = <function> defined @Auctionator\zcUtils.lua:828
 tallyPrint = <function> defined @Auctionator\zcUtils.lua:1145
 msg_yellow = <function> defined @Auctionator\zcUtils.lua:597
 If = <function> defined @Auctionator\zcUtils.lua:110
 Min = <function> defined @Auctionator\zcUtils.lua:80
 msg_str = <function> defined @Auctionator\zcUtils.lua:614
 priceToString = <function> defined @Auctionator\zcUtils.lua:792
 enc64 = <function> defined @Auctionator\zcUtils.lua:226
 Val = <function> defined @Auctionator\zcUtils.lua:58
 PullItemIntoMemory = <function> defined @Auctionator\zcUtils.lua:507
 msg_ex = <function> defined @AddOn

**Comment by ceylina** [comment]
So not sure if this is good lua coding, the right wayto do this, making it primed to explode later or not but it seems that trying to define the length of itemclasses isn't working because there is no table to build something to grab the length of (hopefully my poor explanation skills makes sense - defining itemclasses as a table lets you # it later).

I changed line 352 to this

  local itemClasses = { Atr_GetAuctionClasses() };

and low and behold we got what we needed in #itemclasses and no annoying errors and the code seemed to run as intended in the sell window. Not sure if it will break or not but at least that moved it forward.
